code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,352 @@
+	icon = 'icons/mob/lavaland/lavaland_elites.dmi'
+	faction = list("boss")
+	robust_searching = TRUE
+	ranged_ignores_vision = TRUE
+	ranged = 1
+	obj_damage = 5
+	vision_range = 6
+	aggro_vision_range = 18
+	environment_smash = ENVIRONMENT_SMASH_NONE  //This is to prevent elites smashing up the mining station, we'll make sure they can smash minerals fine below.
+	harm_intent_damage = 0 //Punching elites gets you nowhere
+	stat_attack = UNCONSCIOUS
+	layer = LARGE_MOB_LAYER
+	sentience_type = SENTIENCE_BOSS
+	hud_type = /datum/hud/lavaland_elite
+	var/chosen_attack = 1
+	var/list/attack_action_types = list()
+	var/can_talk = FALSE
+	var/obj/loot_drop = null
+	
+		
+//Gives player-controlled variants the ability to swap attacks
+/mob/living/simple_animal/hostile/asteroid/elite/Initialize(mapload)
+	. = ..()
+	for(var/action_type in attack_action_types)
+		var/datum/action/innate/elite_attack/attack_action = new action_type()
+		attack_action.Grant(src)
+
+//Prevents elites from attacking members of their faction (can't hurt themselves either) and lets them mine rock with an attack despite not being able to smash walls.
+/mob/living/simple_animal/hostile/asteroid/elite/AttackingTarget()
+	if(istype(target, /mob/living/simple_animal/hostile))
+		var/mob/living/simple_animal/hostile/M = target
+		if(faction_check_mob(M))
+			return FALSE
+	. = ..()
+	if(ismineralturf(target))
+		var/turf/closed/mineral/M = target
+		M.gets_drilled()
+		
+//Elites can't talk (normally)!
+/mob/living/simple_animal/hostile/asteroid/elite/say(message, bubble_type, var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
+	if(can_talk)
+		. = ..()
+		return TRUE
+	return FALSE
+		
+/*Basic setup for elite attacks, based on Whoneedspace's megafauna attack setup.
+While using this makes the system rely on OnFire, it still gives options for timers not tied to OnFire, and it makes using attacks consistent accross the board for player-controlled elites.*/
+
+/datum/action/innate/elite_attack
+	name = "Elite Attack"
+	icon_icon = 'icons/mob/actions/actions_elites.dmi'
+	button_icon_state = ""
+	background_icon_state = "bg_default"
+	var/mob/living/simple_animal/hostile/asteroid/elite/M
+	var/chosen_message
+	var/chosen_attack_num = 0
+
+/datum/action/innate/elite_attack/Grant(mob/living/L)
+	if(istype(L, /mob/living/simple_animal/hostile/asteroid/elite))
+		M = L
+		return ..()
+	return FALSE
+
+/datum/action/innate/elite_attack/Activate()
+	M.chosen_attack = chosen_attack_num
+	to_chat(M, chosen_message)
+	
+/mob/living/simple_animal/hostile/asteroid/elite/updatehealth()
+	. = ..()
+	update_health_hud()
+		
+/mob/living/simple_animal/hostile/asteroid/elite/update_health_hud()
+	if(hud_used)
+		var/severity = 0
+		var/healthpercent = (health/maxHealth) * 100
+		switch(healthpercent)
+			if(100 to INFINITY)
+				hud_used.healths.icon_state = "elite_health0"
+			if(80 to 100)
```

Comment:
Only set severity in this switch then below just do icon_state = "elite_health[severity]"